Automattic/wp-calypso
# Add trademark claims notice

#### Changes proposed in this Pull Request

We will be adding some new TLDs that require users to review and agree to a trademark claim notice prior to registering some domains. This PR adds the necessary UI elements to show any trademark claims for certain selected domain names.

Also see https://tools.ietf.org/html/draft-ietf-eppext-tmch-func-spec-01 and
https://newgtlds.icann.org/en/about/trademark-clearinghouse/registries-registrars for more detail.

Depends on D27093-code

#### Testing instructions

- Follow the instructions in D27093-code to make sure you are not using the OTE and that the new TLDs that require TMCH checks are available.
- Build this branch locally or use the calypso.live environment.
- Search for an available domain that has TCN info (or hack the back end to return a false positive for availability checks). For example `dairyqueen.dev`.
- Make sure that when selecting the suggestion, that the TCN screen is shown.
- Make sure that when clicking on the `<- Back` button in the header or the "Choose another domain" button at the bottom of the screen, that you are returned to the domain search page.
- Make sure that when you click on the "Acknowledge Trademark" button that the domain is added to the shopping cart.
- Make sure that the TCN page looks ok on large, medium, and small screen sizes.
- Make sure that the buttons at the bottom of the TCN page are activated only when the notice is scrolled to the bottom (or when the notice is so short that the page isn't scrollable).
- Make sure that the tracks events for `calypso_%_trademark_notice_click` are recorded.
- Make sure that styling looks ok on signup.


You should see something like the following:

![Screen Recording 2019-04-24 at 2 55 20 PM 2019-04-24 15_13_23](https://user-images.githubusercontent.com/1379730/56694759-70da9e00-66b5-11e9-969b-179cdd167a60.gif)

![Screen Recording 2019-04-24 at 3 02 54 PM 2019-04-24 15_11_56](https://user-images.githubusercontent.com/1379730/56694840-a2ec0000-66b5-11e9-9e71-0e99ba681bb2.gif)
